src/common/config_obs_mgr.h

```diff
@@ -38,12 +38,11 @@ class ObserverMgr : public ConfigTracker {
   // This function will assert if you try to delete an observer that isn't
   // there.
   void remove_observer(ConfigObs* observer);
+  void gather_all_observer_changes(config_gather_cb callback);
   template<class ConfigProxyT>
-  void call_all_observers(const ConfigProxyT& proxy);
-  template<class ConfigProxyT>
-  void apply_changes(const std::set<std::string>& changes,
-		     const ConfigProxyT& proxy,
-		     std::ostream *oss);
+  void gather_changes(const std::set<std::string>& changes,
```

Comment:
could you document in the comment how `callback` kicks in in `gather_changes()`?

Reply:
also, i'd suggest rename this method to `for_each_change()`. as it's up to the callback if it wants to "gather" the change or not.